### PR TITLE
Postcode error handling

### PR DESCRIPTION
**Pull request by alexander-griffen**
When the postcode is entered, validation occurs as part of the `clean` function of the form.

The postcode validation is the same as that in CAPE.

Also, validation errors for that field have been added to the html template.

### CODE REVIEW

File: procurement/forms.py [hunk 1]
```diff
@@ -8,3 +10,22 @@ class CouncilChoiceAlertForm(forms.Form):
...
+class HomePageTenderForm(forms.Form):
+    pc = forms.CharField(
+        widget=forms.TextInput(
+            attrs={
+                "type": "search",
+                "name": "pc",
+                "id": "council-search",
+                "class": "form-control",
+            }
+        )
```
Comment (struan) [inline procurement/forms.py]: I don't totally understand why but it looks like the widget stuff here is overridden by the filter so none of this applies. It does seem to work if you move the widget stuff to the filter.

**Review by struan — CHANGES_REQUESTED** [review]
One small issue with the postcode widget but the rest looks good.

The only thing that would be good is it currently just ignores various exceptions in the mapit call and those should probably get handled. Certainly the NotFoundException which is thrown if you put in a postcode that is valid but doesn't exist, e.g. KY34 9QZ

File: procurement/forms.py (hunk 1 above)
Comment (alexander-griffen) [reply procurement/forms.py]: This has now been moved back to the filter, and works

**Comment by alexander-griffen** [comment]
> One small issue with the postcode widget but the rest looks good.
> 
> The only thing that would be good is it currently just ignores various exceptions in the mapit call and those should probably get handled. Certainly the NotFoundException which is thrown if you put in a postcode that is valid but doesn't exist, e.g. KY34 9QZ

This has been achieved - albeit slightly jankily - by moving the logic of the mapit call to the validation step, so that errors can be caught.

File: procurement/views.py [hunk 2]
```diff
@@ -31,6 +31,7 @@ class HomePageView(FilterView):
     def get_queryset(self):
         qs = (
             Tender.objects.all()
+            .filter(state="complete")
```
Comment (struan) [inline procurement/views.py]: There's a typo in this commit message :)

File: procurement/templates/procurement/home.html [hunk 3]
```diff
@@ -8,17 +8,6 @@
...
-        <form class="row" action="" method="GET">
-            <fieldset class="col-sm-auto me-auto">
-                <label for="council-search" class="form-label">Search by postcode or council name</label>
-                <div>{{ filter.form.pc.errors }}</div>
-                <div class="search-input">
-                    <!-- { % include icons/search.html %} -->
-                    {{ filter.form.pc }}
-                </div>
-            </fieldset>
-
```
Comment (struan) [inline procurement/templates/procurement/home.html]: Could probably squash this and the change to `tender_filter` into the previous commit for tidyness.

**Review by struan — APPROVED** [review]
Probably should get the postcode stuff this is based on merged first and then rebase and merge this, but looks good.

**Comment by zarino** [comment]
This branch was merged straight into `master` :+1:

**Pull request closed without merging**